scheme/parsec-cca/evidence_handler.go

```diff
@@ -0,0 +1,418 @@
+package parsec_cca
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/veraison/ear"
+	"github.com/veraison/go-cose"
+	"github.com/veraison/parsec/cca"
+	"github.com/veraison/psatoken"
+	"github.com/veraison/services/handler"
+	"github.com/veraison/services/log"
+	"github.com/veraison/services/proto"
+	"github.com/veraison/services/scheme/common"
+	"github.com/veraison/services/scheme/common/arm"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+const (
+	ScopeTrustAnchor = "trust anchor"
+	ScopeRefValues   = "ref values"
+)
+
+type SwAttr struct {
+	ImplID           []byte `json:"PARSEC_CCA.impl-id"`
+	Model            string `json:"PARSEC_CCA.hw-model"`
+	Vendor           string `json:"PARSEC_CCA.hw-vendor"`
+	MeasDesc         string `json:"PARSEC_CCA.measurement-desc"`
+	MeasurementType  string `json:"PARSEC_CCA.measurement-type"`
+	MeasurementValue []byte `json:"PARSEC_CCA.measurement-value"`
+	SignerID         []byte `json:"PARSEC_CCA.signer-id"`
+	Version          string `json:"PARSEC_CCA.version"`
+}
+
+type CcaPlatformCfg struct {
+	ImplID []byte `json:"PARSEC_CCA.impl-id"`
+	Model  string `json:"PARSEC_CCA.hw-model"`
+	Vendor string `json:"PARSEC_CCA.hw-vendor"`
+	Label  string `json:"PARSEC_CCA.platform-config-label"`
+	Value  []byte `json:"PARSEC_CCA.platform-config-id"`
+}
+
+type Endorsements struct {
+	Scheme  string          `json:"scheme"`
+	Type    string          `json:"type"`
+	SubType string          `json:"subType"`
+	Attr    json.RawMessage `json:"attributes"`
+}
+
+type TaAttr struct {
+	Model    string `json:"PARSEC_CCA.hw-model"`
+	Vendor   string `json:"PARSEC_CCA.hw-vendor"`
+	VerifKey string `json:"PARSEC_CCA.iak-pub"`
+	ImplID   []byte `json:"PARSEC_CCA.impl-id"`
+	InstID   string `json:"PARSEC_CCA.inst-id"`
+}
+
+type TaEndorsements struct {
+	Scheme  string `json:"scheme"`
+	Type    string `json:"type"`
+	SubType string `json:"sub_type"`
+	Attr    TaAttr `json:"attributes"`
+}
+
+type EvidenceHandler struct{}
+
+func (s EvidenceHandler) GetName() string {
+	return "parsec-cca-evidence-handler"
+}
+
+func (s EvidenceHandler) GetAttestationScheme() string {
+	return SchemeName
+}
+
+func (s EvidenceHandler) GetSupportedMediaTypes() []string {
+	return EvidenceMediaTypes
+}
+
+func (s EvidenceHandler) SynthKeysFromRefValue(
+	tenantID string,
+	refVal *proto.Endorsement,
+) ([]string, error) {
+	var (
+		implID string
+		fields map[string]*structpb.Value
+		err    error
+	)
+
+	fields, err = common.GetFieldsFromParts(refVal.GetAttributes())
+	if err != nil {
+		return nil, fmt.Errorf("unable to synthesize reference value abs-path: %w", err)
+	}
+
+	implID, err = common.GetMandatoryPathSegment("PARSEC_CCA.impl-id", fields)
+	if err != nil {
+		return nil, fmt.Errorf("unable to synthesize reference value abs-path: %w", err)
+	}
+
+	lookupKey := arm.RefValLookupKey(SchemeName, tenantID, implID)
+	log.Debug("PARSEC CCA Plugin Reference Value Look Up Key= %s\n", lookupKey)
```

Comment:
this and all the other debug statements below should use the formatted version of the API.  (I'm having a dejavu about this, but I can't remember the details.)

Reply:
this has not been addressed